internal/bbr/manager.go

```diff
@@ -0,0 +1,255 @@
+package bbr
+
+import (
+	"context"
+	"math"
+	"time"
+
+	"capnproto.org/go/capnp/v3/internal/clock"
+)
+
+// A packetMeta contains metadata about a packet that was sent.
+type packetMeta struct {
+	SendTime time.Time // The time at which the packet was sent.
+	Size     int64     // The size of the packet.
+
+	// Whether the connection flow was app-limited when this packet
+	// was sent:
+	AppLimited bool
+
+	// value of corresponding fields in Manager when this packet was sent:
+	Delivered     int64
+	DeliveredTime time.Time
+}
+
+type sendRequest struct {
+	size      int64
+	replyChan chan<- packetMeta
+}
+
+func (m *Manager) StartMessage(ctx context.Context, size uint64) (gotResponse func(), err error) {
+	if size > math.MaxInt64 {
+		panic("TODO: overflow")
+	}
+	replyChan := make(chan packetMeta)
+	select {
+	case <-ctx.Done():
+		return func() {}, ctx.Err()
+	case <-m.ctx.Done():
+		return func() {}, m.ctx.Err()
+	case m.chSend <- sendRequest{size: int64(size), replyChan: replyChan}:
+		pm := <-replyChan
```

Comment:
Are we sure this can never block?  Do we need to select against `ctx.Done()`?

Reply:
Yes we're sure. I added some comments to the chSend field to clarify how this works.